tools/make-unicode-tables.js

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const byline = require('byline');
+const fs = require('fs');
+const http = require('http');
+const path = require('path');
+
+const UNICODE_VERSION = '9.0.0';
+const UCD_LINK = 'http://www.unicode.org/Public/' + UNICODE_VERSION +
+  '/ucd/DerivedCoreProperties.txt';
+const fullTablesFilename = 'unicode_tables.h';
+const rangeTablesFilename = 'unicode_range_tables.h';
+const getHeaderGuard = filename =>
+  `SRC_${filename.replace(/\W/g, '_').toUpperCase()}_`;
+const getOutputPath = filename => path.join(__dirname, '../src', filename);
+
+const getFileHeader = filename =>
+`// Copyright (c) 2017 JSTP project authors. Use of this source code is
```

Comment:
In my opinion it is good to have it here, because it will probably help to avoid any legal problems, given the fact that this file is also a piece of software which works with Unicode data files. Consider this as a precaution, I don't know for sure but I also don't want to find out about it in a bad way.

It helps to understand the resulting format of the output files as well, so that you don't need to look anywhere else or open this auto-generated files (which are pretty big and can slow down some of the editors) to understand their contents.